### ISSUE TITLE
HiScore incorrectly estimates combat

### ISSUE THREAD

**Issue by olligobber** [runelite/runelite#16527]
### Type

Incorrect behavior

### Operating System

Linux

### OS Version

Arch

### Bug description

When checking the high scores of a player who is unranked in some combat skills, the plugin incorrectly calculates their combat level based on only the combat skills they are ranked in. This was kind of addressed in #2039, but the new behaviour still seems incorrect, as it implies too much confidence on the value it shows. Work could be done to improve the estimate, and visually show that it is indeed an estimate or lower bound.

### Screenshots or videos

![2023-04-22-133722_230x438_scrot](https://user-images.githubusercontent.com/11549963/233760124-02a4ab32-1f2f-4686-8ac4-cd28aeae4dc6.png)
![2023-04-22-133715_279x163_scrot](https://user-images.githubusercontent.com/11549963/233760136-6b91b759-ff8e-46bc-8f85-782a9efc27cc.png)

### RuneLite version

RuneLite version 1.9.15.3

Launcher version: 2.6.3

**Comment by olligobber** [comment]
I might look into fixing this myself.

**Comment by Nightfirecat** [comment]
I'm not really sure we can fix this. What would you expect to see here when we don't know any of their combat skill levels besides defense?

**Comment by olligobber** [comment]
Maybe something like "--" which it shows for unranked skills, or maybe ">= 15" to make it more clear it's an estimate.

**Comment by ShayanImran** [comment]
@olligobber @Nightfirecat 
What if we add some extra info to the popout menu to explain that this calculation could be off?

We can maybe add an icon on top of/next to the swords icon to show that there's a warning and prompt a user to hover over it.
<img width="230" alt="image" src="https://user-images.githubusercontent.com/47278487/233897364-b6cf2ada-f10e-4764-b5f6-44ec7de5c6d8.png">

**Comment by GANTR0N** [comment]
What about having a warning symbol or an asterisk next to the potentially inaccurate value? I don't think there's anything wrong with the estimation itself, but informing the user that it is just an estimation could be helpful.

**Comment by Nightfirecat** [comment]
I'm not really understanding what the "warning" or "danger" to users would be. In what cases would it cause concern or issues for users to get an incorrect value from this?

**Comment by ShayanImran** [comment]
Yeah I meant it more so as a notification or indicator that this information is not as accurate with the information we have. And we could have a message explaining that on hover.

**Comment by GANTR0N** [comment]
I don't think there's any sort of danger, so the warning symbol probably wouldn't be ideal to use. An asterisk next to the combat level might be fine, as it would just show that the number is an estimate predicated on incomplete data.

**Comment by piksu** [comment]
Would it make sense to provide an estimate on the combat level? It would be possible to check from the high scores, what is the lowest *visible* level in any given combat skill and provide estimate range based on that.

For example, let's consider a maxed player, with the exception of level 40 defence.

If the lowest appearing visible highscore level for defence is 45, and the looked up player has invisible (uranked) defence (40 in this case), an estimate could provided assuming the player have either 1 or 44 defence.

**Comment by Nightfirecat** [comment]
How would the client know what the lowest level that appears on the hiscores is?

**Comment by piksu** [comment]
> How would the client know what the lowest level that appears on the hiscores is?

The hiscores appears to display results for a total of 2,000,000 accounts. At least in theory, we could write code to check out the rank 2,000,000 player in any particular skill to determine where the line is drawn if someone does not appear in the hiscores for a given skill.

For referece, check out the defence (or any other) hiscore. The page number 80,000 is the last page shown in the hiscores web page.

- https://secure.runescape.com/m=hiscore_oldschool/overall?table=2&page=80000

Defence being an example here, at the time of writing the player holding rank 2,000,000 in defence is level 55 and has 166,643 defence exp.

From here, we could assume if a player have unranked defence level, the level must be between 1 to 55, and more precisely the player holds 1 to 166,642 exp in the skill.

**Comment by Brettyoke49** [comment]
Took a crack at this for funsies. Cheers
